### PR TITLE
Adding correction analysis/visualization.

### PR DESCRIPTION
**Pull request by shixnya**
connectivity.py:
1. Do correction analysis when correction variables are given.
2. 2 bug fixes related to gap junction analysis result assignment.

notebook.py:
1. Additional mode to display corrected matrix
2. Display proability values in cells
3. Mode to show corrected/distance-adjusted to see the effect of the
    correction.
4. RGBA map now respects the range of the values.

### CODE REVIEW

File: aisynphys/ui/notebook.py [hunk 1]
```diff
@@ -196,13 +211,15 @@ def show_connectivity_matrix(ax, results, pre_cell_classes, post_cell_classes, c
...
+                if (show_pmax or correction_only) and np.isfinite(cp):
+                    cprob_str[i,j] += "\n %.3f" %(cp)
             elif ctype == 'electrical':
                 cprob_str[i,j] = "" if result['n_gaps_probed'] == 0 else "%d/%d" % (found, result['n_gaps_probed'])
             cprob_alpha[i,j] = 1.0 - 2.0 * max(cp_upper_ci - cp, cp - cp_lower_ci)
 
     # map connection probability to RGB colors
     mapper = matplotlib.cm.ScalarMappable(norm=norm, cmap=cmap)
-    cprob_rgba = mapper.to_rgba(np.clip(cprob, 0.01, 1.0))
+    cprob_rgba = mapper.to_rgba(np.clip(cprob, norm.vmin, norm.vmax))
```
Comment (campagnola) [inline aisynphys/ui/notebook.py:222]: Maybe we need to make this configurable--there are cases where we want a consistent color range that does not change depending on the data range.

**Comment by campagnola** [comment]
This looks great Shinya! Can you also include your correction model code? Ideally we'd like to provide a way for other people to run the same analysis.

File: aisynphys/ui/notebook.py (hunk 1 above)
Comment (shixnya) [reply aisynphys/ui/notebook.py:222]: Perhaps I confused you in the commit notes.
I meant, 'RGBA map now respects the range the data are clipped'.
This 'norm' variable can be set outside the function (like below) to make the range consistent.

# define a colormap and log normalization used to color the heatmap
norm = matplotlib.colors.LogNorm(vmin=0.01, vmax=1.0, clip=True)
cmap = matplotlib.cm.get_cmap('plasma')

So, as long as you keep this argument the same, the color range would be consistent across figures.

**Comment by shixnya** [comment]
Thanks for taking a look! The correction model is currently written directly in the notebook. Would you like it to be ported to connectivity.py? Also, the current version depends on an external library (iminuit) that is not in 'desktop-environment.yml'. I could remove it if needed, but maybe I'd like to discuss it before committing it.

**Comment by campagnola** [comment]
Yes, I think it should probably go into connectivity.py. To reduce installation headaches, though, let's make iminuit an optional dependency, either by importing it from within the scope where it is used, or by using `optional_import` https://github.com/AllenInstitute/neuroanalysis/blob/master/neuroanalysis/util/optional_import.py#L4

I'm curious, though: does minuit offer any benefits over scipy.optimize or lmfit?

**Comment by shixnya** [comment]
OK. I'll transfer it to connectivity.py, and try to see what the best option is for optimization.

What fitting library should be used was also one of the things I wanted to discuss. Since it's slightly off the topic here, I would make a thread in slack.

I wasn't aware of lmfit, and it seems

**Comment by campagnola** [comment]
Thanks @shixnya !

**Pull request merged**